shinysdr/plugins/basic_demod.py

```diff
@@ -750,6 +750,17 @@ def set_rec_freq(self, freq):
     def get_agc_gain(self):
         return to_dB(self.agc_block.gain())
 
+    @exported_value(
+        type=bool, changes='this_setter', label='AGC Enabled')
```

Comment:
A consequence of this is that if the server is restarted or if a new SSB demodulator is created, the AGC will be immediately disabled but the gain will not be carried over, and so it will have a gain of 1.0 rather than whatever gain was previously locked in.

I'd suggest either making this `persists=False` as a workaround, or making the AGC gain persistent (which also means we can have a free manual gain control).

The latter would just mean writing a setter for `agc_gain`, and maybe renaming/relabeling it since it is no longer strictly automatic.

Reply:
If I write a setter for agc_gain, ideally it would only show as modifiable when AGC is unchecked. Is there a way for a block to widget to change type at runtime? It looks like in basic.js that right now Meter and LinSlider get chosen based on whether there's a setter at all. Otherwise you'll see a slider that can be dragged but is immediately overridden by the AGC when the AGC is enabled.

Reply:
Yes, it's currently not possible to change the writability of a cell over its lifetime. That's definitely something that should become possible eventually, though; `CellMetadata` exists to be a bundle of possibly-changing information (though it can't currently change and writability isn't currently an element of it).

For now, the fact that the slider will be wandering about on its own seems like an adequate cue that it isn't usefully modifiable. Might be good to leave a TODO comment that once we can control writability, it should be used here.

Reply:
Well, that fails with `exceptions.ValueError: persists=True changes=u'continuous' is not allowed`. Any advice or should I just go back to marking it not persistent?

Reply:
…that's true, that doesn't/can't work. Sorry about that. Yeah, go ahead with making agc_enabled non-persistent.